example.py

```diff
@@ -311,7 +543,7 @@ def handle_sigint(_exit_code: int, _frame: FrameType) -> None:
     signal.signal(signal.SIGINT, handle_sigint)
 
     if args.devices:
```

Comment:
This is starting to get a bit complex. It would be cleaner to move the shared code into a utility module and have a separate dump script like https://github.com/python-kasa/python-kasa/blob/master/devtools/dump_devinfo.py so we don't have have the fixture dumping married with the example code.

Reply:
Another example is https://github.com/pyenphase/pyenphase/blob/main/devtools/fixture_collector.py This one is targeted more at users who have new devices so we can collect the fixtures and add them to the test suite to test against new devices

Reply:
Introduced common.py for all tools